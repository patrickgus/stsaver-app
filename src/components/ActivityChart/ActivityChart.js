import React from "react";
import { format } from "date-fns";
import { VictoryChart, VictoryLine, VictoryAxis } from "victory";

function ActivityChart(props) {
  const data = props.logs.map(log => ({
    date:  format(new Date(log.start_time), "MM-dd-yyyy"),
    hours: log.hours
  }));

  return (
    <div className="ActivityChart">
      <VictoryChart domainPadding={10}>
        <VictoryAxis
          tickValues={["02-24-2020", "02-25-2020", "02-26-2020", "03-04-2020"]}
        />
        <VictoryAxis dependentAxis tickFormat={[0, 2, 4, 6, 8, 10]} />
        <VictoryLine data={data} x={"date"} y={"hours"} />
      </VictoryChart>
    </div>
  );
}

export default ActivityChart;
