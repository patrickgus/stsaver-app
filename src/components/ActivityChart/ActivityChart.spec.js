import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import ActivityChart from "./ActivityChart";
import logs from "../../stores/logs-store";

describe("ActivityChart component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <ActivityChart logs={logs} />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the chart as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <ActivityChart logs={logs} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
