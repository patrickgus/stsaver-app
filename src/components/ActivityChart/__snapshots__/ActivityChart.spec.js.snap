// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityChart component renders the chart as expected 1`] = `
<div
  className="ActivityChart"
>
  <div
    className="VictoryContainer"
    style={
      Object {
        "height": "100%",
        "pointerEvents": "none",
        "position": "relative",
        "touchAction": "none",
        "userSelect": "none",
        "width": "100%",
      }
    }
  >
    <svg
      height={350}
      role="img"
      style={
        Object {
          "height": "auto",
          "pointerEvents": "all",
          "width": "100%",
        }
      }
      viewBox="0 0 500 350"
      width={500}
    >
      <g
        role="presentation"
      >
        <line
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "transparent",
              "stroke": "#90A4AE",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeWidth": 2,
            }
          }
          vectorEffect="non-scaling-stroke"
          x1={50}
          x2={450}
          y1={300}
          y2={300}
        />
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={69.81981981981983}
            x2={69.81981981981983}
            y1={300}
            y2={50}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={69.81981981981983}
            x2={69.81981981981983}
            y1={300}
            y2={305}
          />
          <text
            direction="inherit"
            dx={0}
            dy={12.825}
            id="chart-axis-0-tickLabels-0"
            x={69.81981981981983}
            y={313}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 15,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={69.81981981981983}
            >
              Th 3/12
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={250}
            x2={250}
            y1={300}
            y2={50}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={250}
            x2={250}
            y1={300}
            y2={305}
          />
          <text
            direction="inherit"
            dx={0}
            dy={12.825}
            id="chart-axis-0-tickLabels-1"
            x={250}
            y={313}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 15,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={250}
            >
              Fr 3/13
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={430.1801801801802}
            x2={430.1801801801802}
            y1={300}
            y2={50}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={430.1801801801802}
            x2={430.1801801801802}
            y1={300}
            y2={305}
          />
          <text
            direction="inherit"
            dx={0}
            dy={12.825}
            id="chart-axis-0-tickLabels-2"
            x={430.1801801801802}
            y={313}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 15,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={430.1801801801802}
            >
              Sa 3/14
            </tspan>
          </text>
        </g>
      </g>
      <g
        role="presentation"
      >
        <line
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "stroke": "black",
            }
          }
          vectorEffect="non-scaling-stroke"
        />
      </g>
      <text
        direction="inherit"
        dx={15}
        dy={35.325}
        x={0}
        y={0}
      >
        <tspan
          dx={15}
          style={
            Object {
              "fill": "#252525",
              "fontFamily": "'Gill Sans', 'Gill Sans MT', 'SerÂ­avek', 'Trebuchet MS', sans-serif",
              "fontSize": 15,
              "parent": Object {
                "height": "100%",
                "userSelect": "none",
                "width": "100%",
              },
              "stroke": "transparent",
            }
          }
          textAnchor="start"
        >
          Hours
        </tspan>
      </text>
      <g
        role="presentation"
      >
        <path
          d="M 55.53410553410554, NaN 
A 0 0 0 0 0, 55.53410553410554, NaN 
L 55.53410553410554, NaN 
A 0 0 0 0 0, 55.53410553410554, NaN 
L 84.10553410553412, NaN 
A 0 0 0 0 0, 84.10553410553412, NaN 
L 84.10553410553412, NaN 
A 0 0 0 0 0, 84.10553410553412, NaN 
 z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "#455A64",
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 0,
            }
          }
        />
        <path
          d="M 235.71428571428572, NaN 
A 0 0 0 0 0, 235.71428571428572, NaN 
L 235.71428571428572, NaN 
A 0 0 0 0 0, 235.71428571428572, NaN 
L 264.2857142857143, NaN 
A 0 0 0 0 0, 264.2857142857143, NaN 
L 264.2857142857143, NaN 
A 0 0 0 0 0, 264.2857142857143, NaN 
 z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "#455A64",
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 0,
            }
          }
        />
        <path
          d="M 235.71428571428572, NaN 
A 0 0 0 0 0, 235.71428571428572, NaN 
L 235.71428571428572, NaN 
A 0 0 0 0 0, 235.71428571428572, NaN 
L 264.2857142857143, NaN 
A 0 0 0 0 0, 264.2857142857143, NaN 
L 264.2857142857143, NaN 
A 0 0 0 0 0, 264.2857142857143, NaN 
 z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "#455A64",
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 0,
            }
          }
        />
        <path
          d="M 415.8944658944659, NaN 
A 0 0 0 0 0, 415.8944658944659, NaN 
L 415.8944658944659, NaN 
A 0 0 0 0 0, 415.8944658944659, NaN 
L 444.46589446589445, NaN 
A 0 0 0 0 0, 444.46589446589445, NaN 
L 444.46589446589445, NaN 
A 0 0 0 0 0, 444.46589446589445, NaN 
 z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "#455A64",
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 0,
            }
          }
        />
        <path
          d="M 415.8944658944659, NaN 
A 0 0 0 0 0, 415.8944658944659, NaN 
L 415.8944658944659, NaN 
A 0 0 0 0 0, 415.8944658944659, NaN 
L 444.46589446589445, NaN 
A 0 0 0 0 0, 444.46589446589445, NaN 
L 444.46589446589445, NaN 
A 0 0 0 0 0, 444.46589446589445, NaN 
 z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "#455A64",
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 0,
            }
          }
        />
      </g>
    </svg>
    <div
      style={
        Object {
          "height": "auto",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
          "zIndex": 99,
        }
      }
    >
      <svg
        height={350}
        style={
          Object {
            "height": "auto",
            "overflow": "visible",
            "width": "100%",
          }
        }
        viewBox="0 0 500 350"
        width={500}
      />
    </div>
  </div>
</div>
`;
